test/e2e/docker_helper.go

```diff
@@ -202,10 +202,37 @@ func (r *ByoHostRunner) copyKubeconfig(config cpConfig, listopt types.ContainerL
 		kubeconfig = re.ReplaceAll(kubeconfig, []byte("server: https://"+profile.NetworkSettings.Networks[r.NetworkInterface].IPAddress+":6443"))
 	}
 	Expect(os.WriteFile(tempKubeconfigPath, kubeconfig, 0644)).NotTo(HaveOccurred()) // nolint: gosec,gomnd
-
+	listopt.Filters.Add("name", r.ByoHostName)
+	containers, err := r.DockerClient.ContainerList(r.Context, listopt)
+	Expect(err).NotTo(HaveOccurred())
 	config.sourcePath = tempKubeconfigPath
-	config.destPath = r.CommandArgs["--kubeconfig"]
-	err := copyToContainer(r.Context, r.DockerClient, config)
+	// get the $HOME env variable to set the destination path for kubeconfig
+	execCommand, err := r.DockerClient.ContainerExecCreate(r.Context, containers[0].ID, types.ExecConfig{
+		AttachStdin:  false,
+		AttachStdout: true,
+		AttachStderr: true,
+		Cmd:          []string{"sh", "-c", "echo ${HOME}"},
+	})
+	Expect(err).ShouldNot(HaveOccurred())
+	resp, err := r.DockerClient.ContainerExecAttach(r.Context, execCommand.ID, types.ExecStartCheck{})
+	Expect(err).ShouldNot(HaveOccurred())
+	defer resp.Close()
+	homeDir, err := resp.Reader.ReadString('\n')
+	Expect(err).ShouldNot(HaveOccurred())
+	homeDir = strings.TrimSuffix(homeDir, "\n")
+	config.destPath = strings.SplitAfterN(strings.TrimSpace(homeDir)+"/.byoh/config", "/", 2)[1] // nolint: gomnd
```

Comment:
Just curious why do we need to split here?

Reply:
good catch, reading the container exec response in `resp.Reader.ReadString('\n')` has some special characters in the prefix which were not even logged in `fmt.Print`. The output I was expecting is `/root/`

Reply:
okay 👍  Can we have comment on the line to increase readability.